src/org/traccar/protocol/AdmProtocolDecoder.java

```diff
@@ -66,7 +66,7 @@ protected Object decode(
             position.setProtocol(getProtocolName());
             position.setDeviceId(deviceSession.getDeviceId());
 
-            buf.readUnsignedByte(); // firmware version
+            position.set(Position.KEY_VERSION_FW, buf.readUnsignedByte()); // firmware version
```

Comment:
Please remove comments like this when you are implementing parameters. KEY_VERSION_FW is clear enough. There is no need to have a comment. Applies to all your changes.